engine/schema/src/com/cloud/vm/dao/DomainRouterDaoImpl.java

```diff
@@ -432,4 +432,12 @@ public void removeRouterFromGuestNetwork(final long routerId, final long guestNe
         sc.setParameters("states", State.Stopped);
         return listBy(sc);
     }
+
+    @Override
+    public List<DomainRouterVO> listIncludingRemovedByVpcId(long vpcId) {
+        SearchCriteria<DomainRouterVO> sc = VpcSearch.create();
```

Comment:
QueryBuilder is a better abstraction to use for building queries it saves the hardcoding of column names.